File: solutions/javascript/house/1/house.js
export class House {   
  
  static verse(num) {    
     const objects = ["the house that Jack built.", "the malt", "the rat", "the cat", "the dog", "the cow with the crumpled horn", "the maiden all forlorn", "the man all tattered and torn", "the priest all shaven and shorn", "the rooster that crowed in the morn", "the farmer sowing his corn","the horse and the hound and the horn"];

    const verbs = ["that lay in", "that ate", "that killed", "that worried", "that tossed", "that milked", "that kissed", "that married", "that woke", "that kept", "that belonged to"];

    const intro = "This is ";
    var res = [intro + objects[num-1]];
    if (num>1) {
      for(var idx=num-1; idx>0; idx--){
        let str = verbs[idx-1] + " " + objects[idx-1];
        res.push(str);
      }
    }
    return res;  
  }

  static verses(first, last) {
    let out=[];
    for(var n = first; n<=last; n++){
        out = out.concat(this.verse(n));
        if (n<last) {
          out.push('');
        }
    }
    return out;    
  }
}

File: solutions/dart/house/1/lib/house.dart
class House {

 static  List<String> objects = ["house that Jack built.", "malt", "rat", "cat", "dog", "cow with the crumpled horn", "maiden all forlorn", "man all tattered and torn", "priest all shaven and shorn", "rooster that crowed in the morn", "farmer sowing his corn","horse and the hound and the horn"];

  static List<String> verbs =  ["lay in", "ate", "killed", "worried", "tossed", "milked", "kissed", "married", "woke", "kept", "belonged to"]; 

  static String intro = 'This is ';


  String verse(int num){
      var res = intro + 'the ' + objects[num-1];
      for(var i=num-1; i>0; i--){
        res += ' that ' + verbs[i-1] + ' the ' + objects[i-1];
      }
      return res;    
  }

  String recite(int from, int to){
    List<String> verses = [];
    for(var i=from; i<=to; i++){
      verses.add(verse(i));  
      if(i<to)
        verses.add('\n');
    }
    return verses.join('');
  }
}
